### ISSUE TITLE
AKAO PSF missing most instruments (but multisampled instruments present)

### ISSUE THREAD

**Issue by nyanpasu64** [vgmtrans/vgmtrans#126]
## Expected Behavior

When ripping AKAO PSF files, all instruments are ripped properly, to their correct instrument numbers.

## Current Behavior

Only instruments containing multiple samples are ripped. Instruments containing one sample are missing.

## Steps to Reproduce (for bugs)

1. Load "Chrono Cross Dragon God PSF"
2. Save as MIDI and SoundFont 2.

## Context
Ripping samples, using SF2 tuning information for SNES BRR tuning, listening to the MIDI to identify which tracks contribute to the sound.

## Your Environment

* VGMTrans version used: Longstanding bug present in Git master
* Operating System and version: Windows 10 x64
* Game ROM SHA-256 or music dump release name (Zophar, VGMRips, etc): ???

**Comment by jimbo1qaz** [comment]
I tried extracting DLS, then converting to SF2. Issue was fixed.

**Comment by nyanpasu64** [comment]
I tried the song again on c251160f9d9c126d4ada8e32f4e8c66051985e12. It... almost works [(MIDI/SF2/DLS download)](https://github.com/vgmtrans/vgmtrans/files/5244168/dragon.god.midi.sf2.dls.zip). I inspected the dls/sf2 in Awave Studio and the MIDI in Sekaiju.

The .dls maps multi-sampled instruments and the drum kit to a nonzero CC32. (IDK if .dls stores CC0 and CC32 separately.)

The .sf2 maps multi-sampled instruments and the drum kit to a nonzero CC0... To be precise, .sf2 only stores a single bank number, which both Awave Studio and BASSMIDI (used by OmniMIDI) interprets as CC0. IDK which is "correct", strictly speaking.

The issue is, vgmtrans writes MIDIs with nonzero CC32, causing the instruments to not be loaded properly on BASSMIDI.

The confounding issue is that the PS1 has 24 channels, and the resulting MIDI file has 23 tracks each mapped to a MIDI channel (unsure if all are monophonic), but MIDI files/protocol only has 16 channels (only 15 of which are used by vgmtrans). As a result, tracks assigned to the same channel overwrite each other's instrument mappings in practically all players. The only way I was even able to diagnose the bank-mapping problem was by telling Sekaiju to solo individual tracks.

Is anyone aware of a "MIDI type 1 file" player which isolates each track even if it shares a channel with another track? I wrote a command-line player many years ago, but don't have a build ready to use now.

Should vgmtrans map instrument 127 onto MIDI channel 9=10, during sf2/mid/dls export?

External links:

- BassMIDI MIDI/SF2, CC0 vs CC32: https://www.hcs64.com/mboard/forum.php?showthread=45098&showpage=4#post_45541
- Discussion of MIDI/SF2, CC0 and CC32: https://lists.gnu.org/archive/html/fluid-dev/2013-11/msg00030.html
